### PR TITLE
UPSTREAM: 58533: add suggestion to describe pod for container names

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1389208

Usability improvement: Suggest using `oc describe <pod>` to retrieve all pod containers to use with cmds that stream to/from pods.

cc @soltysh

### CODE REVIEW

**Comment by juanvallejo** [comment]
Upstream: https://github.com/kubernetes/kubernetes/pull/58533

**Comment by soltysh** [comment]
Or at least tagged for merge.

**Comment by deads2k** [comment]
/hold

until upstream is in queue so we can figure out what to review and what not to review

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/attach.go [hunk 1]
```diff
@@ -312,6 +322,10 @@ func (p *AttachOptions) containerToAttachTo(pod *api.Pod) (*api.Container, error
 		return nil, fmt.Errorf("container not found (%s)", p.ContainerName)
 	}
 
+	if len(p.SuggestedCmdUsage) > 0 {
+		fmt.Fprintf(p.Err, "%s\n", p.SuggestedCmdUsage)
```
Comment (soltysh) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/attach.go]: You need to update this to match upstream.

**Comment by juanvallejo** [comment]
@deads2k 
updated upstream changes to match merged upstream PR.
updated Origin changes to ensure defaulted container name is printed

**Pull request merged**